### PR TITLE
Fix dataflow and mlengine samples with recent changes

### PR DESCRIPTION
**Pull request by hongye-sun**
The changes are:
* Dataflow interface changes
* mlengine-train add job_dir as output
* add type info

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1006)

### CODE REVIEW

File: components/gcp/dataflow/launch_template/README.md [hunk 1]
```diff
@@ -69,16 +69,26 @@ import json
...
-    validate_only='', 
-    wait_interval = 30
-):
-    dataflow_template_op(project_id, gcs_path, launch_parameters, location, job_name_prefix, validate_only, 
-        wait_interval).apply(gcp.use_gcp_secret('user-gcp-sa'))
+    project_id = PROJECT_ID, 
+    gcs_path = 'gs://dataflow-templates/latest/Word_Count', 
+    launch_parameters = json.dumps({
+       'parameters': {
+           'inputFile': 'gs://dataflow-samples/shakespeare/kinglear.txt',
```
Comment (gaoning777) [inline components/gcp/dataflow/launch_template/README.md]: In which project is the file stored? 
I want to make sure we keep a good track of the released dataset.

Comment (hongye-sun) [reply components/gcp/dataflow/launch_template/README.md]: The data is from dataflow samples. Dataflow team is maintaining that. I will add more description in the notebook on how to build a dataflow beam code and will have reference to dataflow documentation on the sample code and data.

Comment (gaoning777) [reply components/gcp/dataflow/launch_template/README.md]: SG

**Pull request merged**